pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -145,12 +145,20 @@ def restore_model(self) -> None:
     def restore_model_weights(self, checkpoint_path: Optional[Union[str, Path]]) -> None:
         """ Restore only the model weights. """
         checkpoint = self._loaded_checkpoint
+        if hasattr(self.trainer.training_type_plugin, "serialized_restore_model_state"):
+            checkpoint = self.trainer.training_type_plugin.serialized_restore_model_state(checkpoint_path)
+        else:
+            checkpoint = self.restore_model_state(checkpoint_path)
+
+    def restore_model_state(self, checkpoint_path: Optional[Union[str, Path]]) -> dict:
```

Comment:
maybe it's better to just incorporate it into the above if else. 
a separate method with a similar name is maybe a bit confusing?